Kiosk/App/CardHandler.swift

```diff
@@ -1,107 +1,181 @@
 import UIKit
 import RxSwift
+import CardFlight
 
-class CardHandler: NSObject, CFTReaderDelegate {
+class CardHandler: NSObject, CFTTransactionDelegate {
 
-    fileprivate let _cardStatus = PublishSubject<String>()
+    private let _cardStatus = PublishSubject<String>()
+    private let _userMessages = PublishSubject<String>()
+    private var cardReader: CFTCardReaderInfo?
+
+    var transaction: CFTTransaction?
 
     var cardStatus: Observable<String> {
         return _cardStatus.asObservable()
     }
+    // TODO: Surface these messages to the user in the Kiosk UI.
+    var userMessages: Observable<String> {
+        // User messages are things like "Swipe card", "processing", or "Swipe card again". Due to a problem with the
+        // CardFlight SDK, the user is prompted to accept processing for card tokenization, which is provides a
+        // unfriendly user experience (prompting to accept a transaction that we're not actually placing). So we
+        // auto-accept these requests and filter out confirmation messages, which don't apply to tokenization flows,
+        // until this issue is fixed: https://github.com/CardFlight/cardflight-v4-ios/issues/4
```

Comment:
Hey @orta, when you have a moment can you double-check my thinking here? I need to make sure this is not sneaky or unethical, and make sure that the intentions of the code are made clear in the comment for other developers. Thanks!

Reply:
Yeah, this makes sense for me